### PR TITLE
feat: add saml group links

### LINKED ISSUE #2270: Add support for API Endpoint for SAML Group Sync
## Description of the problem, including code/CLI snippet
GitLab added the API Endpoint for SAML Group Sync in 15.3. Please add support for this new API endpoint.

## Expected Behavior
Supports SAML Group Sync

## Actual Behavior
Not yet supported

## Specifications

  - python-gitlab version:
  - API version you are using (v3/v4):
  - Gitlab server version (or gitlab.com):

### PR DESCRIPTION
**Pull request by abhiandthetruth**
CLOSES #2270

### CODE REVIEW

**Comment by nejch** [comment]
Just ping us here if you get stuck with anything @abhiandthetruth. Thanks for working on this!

**Comment by abhiandthetruth** [comment]
@nejch occupied with some other work atm. Will try to complete soon

**Comment by abhiandthetruth** [comment]
@nejch can you have a look?
Disabled functional tests since cant get them to work, getting 401 while creating a new saml group link. Might need to setup an identity provider in CI.

File: tests/functional/api/test_groups.py [hunk 1]
```diff
@@ -304,3 +304,12 @@ def test_group_transfer(gl, group):
 
     transferred_group = gl.groups.get(transfer_group.id)
     assert transferred_group.path == transferred_group.full_path
+
+
+@pytest.mark.gitlab_premium
+@pytest.mark.xfail(reason="need to setup an identity provider or it's mock")
+def test_group_saml_group_links(gl):
+    group = gl.groups.create({"name": "gitlab-test-group", "path": "gitlab-test-group"})
```
Comment (lmilbaum) [inline tests/functional/api/test_groups.py:312]: Does it worth to add a check that the group was created? Maybe there is a rase condition here.

Comment (nejch) [reply tests/functional/api/test_groups.py:312]: Actually, I'd suggest reusing our `group` fixture and not creating groups inside the test case. Unless there was a reason to create a new one @abhiandthetruth?

```diff
-    def test_group_saml_group_links(gl):
-    group = gl.groups.create({"name": "gitlab-test-group", "path": "gitlab-test-group"})
+    def test_group_saml_group_links(group):
```

See for example the LDAP group links tests:
https://github.com/python-gitlab/python-gitlab/blob/863ea11e2fbea3273c9177eb04df00a37bac6044/tests/functional/api/test_groups.py#L147-L172

Comment (abhiandthetruth) [reply tests/functional/api/test_groups.py:312]: ohh this. Sorry I was just testing if every method is giving 401 in this particular test lol. So tried creating a group in the functional test. Was using the fixture initially. Will fix, thanks!

Comment (abhiandthetruth) [reply tests/functional/api/test_groups.py]: @lmilbaum @nejch fixed!

**Comment by nejch** [comment]
> @nejch can you have a look? Disabled functional tests since cant get them to work, getting 401 while creating a new saml group link. Might need to setup an identity provider in CI.

note to self:
```
# https://docs.gitlab.com/ee/integration/omniauth.html#configure-initial-settings
# https://docs.gitlab.com/ee/integration/saml.html#general-setup
gitlab_rails['omniauth_allow_single_sign_on'] = ['saml']
gitlab_rails['omniauth_block_auto_created_users'] = false
gitlab_rails['omniauth_providers'] = [
  {
    name: "saml",
    args: {
      assertion_consumer_service_url: "http://localhost:8080/users/auth/saml/callback",
      idp_cert_fingerprint: "43:51:43:a1:b5:fc:8b:b7:0a:3a:a9:b1:0f:66:73:a8",
      idp_sso_target_url: "https://login.example.com/idp",
      issuer: "http://localhost:8080",
      name_identifier_format: "urn:oasis:names:tc:SAML:2.0:nameid-format:persistent"
    }
  }
]
```

**Review by nejch — APPROVED** [review]
Thanks for the work here @abhiandthetruth! The request body matcher in the unit tests will really help with confidence until we have a functional test setup for this.

I've squashed the commits here as the others weren't strictly for the changelog as a `feat` :)

**Pull request merged**